tests/functional/ui/test_odf_storage_consumption_trend.py

```diff
@@ -0,0 +1,314 @@
+        # Below sleep is mandatory for mgr failover, if not the same pod will become active again.
+        time.sleep(60)
+        logger.info(f"Scale down {active_mgr_deployment_name} to 1")
+        helpers.modify_deployment_replica_count(
+            deployment_name=active_mgr_deployment_name, replica_count=1
+        )
+        assert (
+            len(get_mgr_pods()) == 2
+        ), "one of the mgr pod is still down after scale down and up"
+        logger.info("Mgr failovered successfully")
+
+        logger.info("Now the testing will begin for consumption trend UI")
+        est_days = self.get_est_days_from_ui()
+        average = self.get_avg_consumption_from_ui()
+        logger.info(f"From the UI, Estimated Days: {est_days} and Average: {average}")
+        days_avg_tpl = self.calculate_est_days_and_average_manually()
+        first_validation = est_days == days_avg_tpl[0]
+        # rel_tol 0.1 means upto 10% tolerance, which means that difference between  manual and UI est days is up to 10%
+        second_validation = math.isclose(est_days, days_avg_tpl[0], rel_tol=0.1)
+        if first_validation:
+            logger.info(
+                "Manually calculated and UI values are exactly matched for 'Estimated days until full'"
+            )
+        elif second_validation:
+            logger.warning(
+                "Manually calculated and UI values are matched with upto 10% difference for 'Estimated days until full'"
+            )
+        assert (
+            first_validation or second_validation
+        ), "'Estimated days until full' is wrongly displayed in UI"
+
+    def test_consumption_trend_with_prometheus_failures(self, setup_ui_class):
+        """
+        Fail prometheus and verify the Consumption trend in the ODF dashboard to make sure
+        ‘Estimated days until full’ and 'Average' reflects accurate value.
+
+            1. Fail the prometheus pods by deleting them. New pods will be created automatically.
+            2. When the new prometheus pods  up, Consumption trend should be displayed in the dashboard.
+            3. Check the values for  ‘Estimated days until full’ and ‘Average of storage consumption per day’
+            4. Should show close to the values before deleting the prometheus pod
+
+        """
+        logger.info("Get the value of 'Estimated days until full' from UI")
+        est_days_before = self.get_est_days_from_ui()
+        logger.info(
+            f"The value of 'Estimated days until full' from UI is {est_days_before} before failing prometheus"
+        )
+        average_before = self.get_avg_consumption_from_ui()
+        logger.info(
+            f"'Average of storage consumption per day' from UI is {average_before} before failing prometheus"
+        )
+        logger.info("Bring down the prometheus")
+        list_of_prometheus_pod_obj = get_prometheus_pods()
+        delete_pods(list_of_prometheus_pod_obj)
+        time.sleep(120)
```

Comment:
we need to add flexible timeouts using TimeoutSampler fetching the state instead of static sleeps

Reply:
done